pkg/manager/manager.go

```diff
@@ -161,6 +163,24 @@ func (j Job) IsComplete() bool {
 	return j.Complete || len(j.Credentials) > 0 || (len(j.State) > 0 && j.State != prowapiv1.PendingState)
 }
 
+type WorkflowConfig struct {
+	Workflows map[string]WorkflowConfigItem `yaml:"workflows"`
+	Mutex     sync.RWMutex                  `yaml:"-"` // this field just allows us to update the above values without races
+}
+
+type WorkflowConfigItem struct {
+	BaseImages   map[string]citools.ImageStreamTagReference `yaml:"base_images,omitempty"`
+	Architecture string                                     `yaml:"architecture,omitempty"`
+	Platform     string                                     `yaml:"platform"`
+}
+
+// LeaseClient only include the metrics function, as we don't want to create leases
+type LeaseClient interface {
+	// Metrics queries the states of a particular resource, for informational
+	// purposes.
+	Metrics(rtype string) (lease.Metrics, error)
+}
+
```

Comment:
Since we are refactoring into packages, it's probably best to introduce a `pkg/manager/types.go` instead of carrying everything around hidden in all the nooks and crannies.

Reply:
All types are moved to `types.go`.